diff --git a/config.def.h b/config.def.h
index e06be66..f2121d2 100644
--- a/config.def.h
+++ b/config.def.h
@@ -4,7 +4,7 @@
 const unsigned int interval = 1000;
 
 /* text to show if no value can be retrieved */
-static const char unknown_str[] = "n/a";
+static const char unknown_str[] = "✗";
 
 /* maximum output string length */
 #define MAXLEN 2048
@@ -62,6 +62,10 @@ static const char unknown_str[] = "n/a";
  * wifi_essid          WiFi ESSID                      interface name (wlan0)
  */
 static const struct arg args[] = {
-	/* function format          argument */
-	{ datetime, "%s",           "%F %T" },
+	/* function format	string			argument	*/
+	{ battery_state,	"%s",			"BAT0"			},
+	{ battery_perc, 	"B %s%%   ",	"BAT0"			},
+	{ wifi_essid, 		"%s ",			"wlan0" 		},
+	{ wifi_perc,		"%s%%   ",		"wlan0" 		},
+	{ datetime, 		"%s",			"%a %d %b %R" 	}
 };
diff --git a/slstatus.h b/slstatus.h
index 08f610a..7373863 100644
--- a/slstatus.h
+++ b/slstatus.h
@@ -74,8 +74,8 @@ const char *username(void);
 const char *uid(void);
 
 /* volume */
-const char *vol_perc(const char *card);
+const char *vol_perc(const char *);
 
 /* wifi */
-const char *wifi_perc(const char *interface);
-const char *wifi_essid(const char *interface);
+const char *wifi_perc(const char *);
+const char *wifi_essid(const char **);
